container-engine-lib/lib/backend_impls/kubernetes/object_attributes_provider/logs_collector_object_attributes_provider.go

```diff
@@ -0,0 +1,103 @@
+package object_attributes_provider
+
+import (
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/kubernetes/object_attributes_provider/kubernetes_annotation_key"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/kubernetes/object_attributes_provider/kubernetes_annotation_value"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/kubernetes/object_attributes_provider/kubernetes_label_key"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/kubernetes/object_attributes_provider/kubernetes_label_value"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_impls/kubernetes/object_attributes_provider/label_value_consts"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/logs_collector"
+	"github.com/kurtosis-tech/stacktrace"
+)
+
+const (
+	logsCollectorNamePrefix       = "kurtosis-logs-collector"
+	logsCollectorConfigNamePrefix = "kurtosis-logs-collector-config"
+)
+
+type KubernetesLogsCollectorObjectAttributesProvider interface {
+	ForLogsCollectorDaemonSet() (KubernetesObjectAttributes, error)
+
+	ForLogsCollectorNamespace() (KubernetesObjectAttributes, error)
+
+	ForLogsCollectorConfigMap() (KubernetesObjectAttributes, error)
+
+	// TODO: might need to implement these if the log collector requires roles for requesting tags from api server
+	//ForLogsCollectorServiceAccount() (KubernetesObjectAttributes, error)
+	//ForLogsCollectorClusterRole() (KubernetesObjectAttributes, error)
+	//ForLogsCollectorClusterRoleBindings() (KubernetesObjectAttributes, error)
+}
+
+func GetKubernetesLogsCollectorObjectAttributesProvider(logsCollectorGuid logs_collector.LogsCollectorGuid) KubernetesLogsCollectorObjectAttributesProvider {
+	return newKubernetesLogsCollectorObjectAttributesProvider(logsCollectorGuid)
+}
+
+type kubernetesLogsCollectorObjectAttributesProviderImpl struct {
+	logsCollectorGuid logs_collector.LogsCollectorGuid
+}
+
+func newKubernetesLogsCollectorObjectAttributesProvider(logsCollectorGuid logs_collector.LogsCollectorGuid) *kubernetesLogsCollectorObjectAttributesProviderImpl {
+	return &kubernetesLogsCollectorObjectAttributesProviderImpl{
+		logsCollectorGuid: logsCollectorGuid,
+	}
+}
+
+func (provider *kubernetesLogsCollectorObjectAttributesProviderImpl) ForLogsCollectorDaemonSet() (KubernetesObjectAttributes, error) {
+	name, err := getCompositeKubernetesObjectName([]string{logsCollectorNamePrefix, string(provider.logsCollectorGuid)})
+	if err != nil {
+		return nil, err
+	}
+
+	labels := map[*kubernetes_label_key.KubernetesLabelKey]*kubernetes_label_value.KubernetesLabelValue{
+		kubernetes_label_key.KurtosisResourceTypeKubernetesLabelKey: label_value_consts.LogsCollectorKurtosisResourceTypeKubernetesLabelValue,
```

Comment:
I considered giving the logs collector a guid but made it through with just this label being sufficient for managing the logs collector retrieval/creation/removal - LMK if you think there's any other good reason to add the guid

Reply:
Only reason I can think of is a creation-deletion cycle that happens quickly, which could end up with grabbing a still-deleting collector and then the new one. But not sure if that's a real risk